File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/Fruit.java
package org.acme.hibernate.orm.panache;

import jakarta.persistence.Cacheable;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;

import io.quarkus.hibernate.reactive.panache.PanacheEntity;

@Entity
@Cacheable
public class Fruit extends PanacheEntity {

    @Column(length = 40, unique = true)
    public String name;

    public Fruit() {
    }

    public Fruit(String name) {
        this.name = name;
    }

}

File: hibernate-orm-panache-kotlin-quickstart/src/main/kotlin/org/acme/hibernate/orm/panache/Fruit.kt
package org.acme.hibernate.orm.panache

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import jakarta.persistence.Cacheable
import jakarta.persistence.Column
import jakarta.persistence.Entity

@Entity
@Cacheable
class Fruit : PanacheEntity {
    companion object: PanacheCompanion<Fruit>
    @Column(length = 40, unique = true)
    lateinit var name: String

    constructor()

    constructor(name: String) {
        this.name = name
    }
}